### PR TITLE
Add generic auth

### PR DESCRIPTION
**Pull request by garrettjoecox**
## Highlights:
- Add Yup & Formik
- Add authMock to exampleService
- Add authSlice (with refreshAuth, login, and logout thunks)
- Add LandingNavigator, LoginScreen, LoginForm
- Add logout button to home screen
- Various changes to navigation setup to account for auth flow
- Pressing "don't have an account?" Prompts them that this is a private beta and signups are closed

## Screenshots:
Screen Shot 2021-03-05 at 11 37 11 AM
Screen Shot 2021-03-05 at 11 37 27 AM
Screen Shot 2021-03-05 at 11 37 38 AM

<details>
<summary>File System Diff</summary>

```diff
@@ -61,19 +61,26 @@
   |-tsconfig.json
   |-.prettierrc.js
   |-src
   |  |-config
   |  |  |-index.ts
   |  |-domains
+  |  |  |-landing
+  |  |  |  |-components
+  |  |  |  |  |-LoginForm.tsx
+  |  |  |  |-screens
+  |  |  |  |  |-LandingNavigator.tsx
+  |  |  |  |  |-LoginScreen.tsx
   |  |  |-core
   |  |  |  |-hooks
   |  |  |  |  |-useNavigationRestore.ts
   |  |  |  |-App.tsx
   |  |  |  |-screens
   |  |  |  |  |-ModalNavigator.tsx
   |  |  |  |  |-BottomTabNavigator.tsx
+  |  |  |  |  |-SplashScreen.tsx
   |  |  |  |  |-AppNavigator.tsx
   |  |  |-todo
   |  |  |  |-components
   |  |  |  |  |-Checkbox.tsx
   |  |  |  |  |-Todo.tsx
   |  |  |  |-screens
@@ -91,16 +98,18 @@
   |  |  |  |-typography.ts
   |  |  |  |-index.ts
   |  |  |  |-colors.ts
   |  |-types
   |  |  |-theme.d.ts
   |  |-state
+  |  |  |-authSlice.ts
   |  |  |-index.ts
   |  |  |-todoSlice.ts
   |  |-services
   |  |  |-exampleService
+  |  |  |  |-authMock.ts
   |  |  |  |-getExample.ts
   |  |  |  |-client.ts
   |  |  |  |-types.ts
   |  |  |  |-index.ts
   |  |  |-todo
   |  |  |  |-api.ts
```
</details>

### CODE REVIEW

File: src/domains/home/HomeScreen.tsx [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
-export const HomeScreen: FC = () => (
-  <Screen variant="centered">
-    <Text fontWeight="black">Hello World</Text>
-  </Screen>
-);
+export const HomeScreen: FC = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const onPressLogout = useCallback(() => {
+    dispatch(logout());
+  }, [dispatch]);
```
Comment (murtyjones) [inline src/domains/home/HomeScreen.tsx:14]: i suspect technically `logout` should be a dependency but i doubt it matters

File: src/domains/landing/components/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import { Input } from 'src/components/Input';
+import { PasswordInput } from 'src/components/PasswordInput';
+import { Text } from 'src/components/Text';
+import { AppDispatch } from 'src/state';
+import { login } from 'src/state/authSlice';
+import * as Yup from 'yup';
+
+const VALIDATION_SCHEMA = Yup.object({
+  email: Yup.string().email('Email is invalid').required('Email is required'),
+  password: Yup.string().max(255, 'Password too long').required('Password is required'),
```
Comment (murtyjones) [inline src/domains/landing/components/LoginForm.tsx]: i like yup's format

Comment (murtyjones) [reply src/domains/landing/components/LoginForm.tsx]: password min length?

**Review by murtyjones — APPROVED** [review]
one suggestion on PW requirements but LGTM!

File: src/domains/landing/components/LoginForm.tsx (hunk 2 above)
Comment (garrettjoecox) [reply src/domains/landing/components/LoginForm.tsx]: Technically I should probably remove the max, doesn't make sense to restrict this on login (just on signup)

File: src/domains/home/HomeScreen.tsx (hunk 1 above)
Comment (garrettjoecox) [reply src/domains/home/HomeScreen.tsx:14]: Logout is declared outside of the of scope of the component, so the dep wouldn't be neceessary

**Pull request merged**